rfcs/0000-kuadrant_crd.md

```diff
@@ -0,0 +1,164 @@
+`limitador-operator`. Disabling the `AuthPolicy` yields the same result for the `Authorino` object.
+Additionally it will avoid registering the [external
+authorizer](https://istio.io/latest/docs/tasks/security/authorization/authz-custom/) with Istio. The
+both policies API will be exposed and the reconcilers will still be registered. Creating a policy that
+is disabled would be denied.
+
+A disabled policy can be enabled by changing the value of the `policy` key. Flipping the value can
+only happen when no policy object exist, e.g. in order to disable the `authentication`, no
+`AuthPolicy` can exist or this would fail.
+
+While no default `Limitador` or `Authorino` object would be created, a user can absolutely create one
+or the other and use their APIs directly. In the case of authentication, the user would need to
+register the external authorizer with Istio themselves.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Changing the definition of the `kuadrant` custom resource is relatively straight forward. It is
+intended that the way to enable or disable the support for either policy is not a boolean value. Using
+an enumeration, i.e. `disabled` or `enabled`, enables growing the set of values later.
+
+## Installation
+
+On initial installation, honoring one or the other setting would be fairly straight forward as well:
+
+The both CRDs, as well as their reconcilers all get registered whether the `policy` support is
+`enabled` or `disabled`.
+
+- `enabled` The Kuadrant controller makes sure the external authorizer gets registered with Istio and
+  makes sure there is an `Authorino` and/or `Limitador` object without any changes as it does today.
+- `disabled` None of the above would happen. Most importantly, any missing part required for the above
+  shouldn't have the reconciliation fail and prevent further work it requires to do (as reconciling
+  another policy that's being `enabled`).
+
+Additionally a `ValidatingWebhookConfiguration` gets registered for each CR's `CREATE` operation for
+the entire cluster. The webhook will only allow admissions of resources for which the policy is
+currently enabled. `UPDATE` and `DELETE` should remain unhindered, as in order to transition the
+policy support requires no policies to be present.
+
+If any of the following steps fails, all previously applied changes need to be unrolled and the
+`status` block of the CR would clearly reflected what changed and that Kuadrant is currently _not_
+operational. This is to be considered an atomic operation, either all gets successfully deployed or
+nothing is.
+
+If a policy is disabled but a policy is being added, when disallowing the admission, the webhook would
+use a response code `501 Not Implemented` with the additional `message` indicating the policy is
+disabled and as such can't be honored.
+
+### In an OSSM cluster
+
+The deployment of `Kuadrant` should fail if `authentication` isn't explicitly disabled.
```

Comment:
should we make this explicit in this proposal? I assume we will want it to succeed in the future?

Reply:
@eguzki's right. This section can now be removed from the RFC.